prospero-cli/src/main/java/org/wildfly/prospero/cli/commands/UpdateCommand.java

```diff
@@ -471,4 +478,49 @@ public static Path detectProsperoInstallationPath() throws ArgumentParsingExcept
         return Paths.get(modulePath).toAbsolutePath().getParent();
     }
 
+
+    public static void compareManifests(List<ManifestVersionRecord.MavenManifest> mavenManifests, List<Artifact> manifestUpdates ) {
+
+        List<ManifestVersionRecord.MavenManifest> intersectingManifests = new ArrayList<>();
+
+        // Find intersecting manifests (those that are present in both mavenManifests and manifestUpdates)
+        for (ManifestVersionRecord.MavenManifest installedManifest : mavenManifests) {
+            for (Artifact update : manifestUpdates) {
+                if (update.getGroupId().equals(installedManifest.getGroupId()) &&
+                        update.getArtifactId().equals(installedManifest.getArtifactId())) {
+                    intersectingManifests.add(installedManifest); // Add to intersection list
+                    break;
+                }
+            }
+        }
+
+        // For each intersecting manifest, compare the version
+        for (ManifestVersionRecord.MavenManifest installedManifest : intersectingManifests) {
+            Artifact updateArtifact = null;
+
+            // Find the corresponding update artifact for the installedManifest
+            for (Artifact manifestUpdate : manifestUpdates) {
+                if (manifestUpdate.getGroupId().equals(installedManifest.getGroupId()) &&
+                        manifestUpdate.getArtifactId().equals(installedManifest.getArtifactId())) {
+                    updateArtifact = manifestUpdate;
+                    break;
+                }
+            }
+
+            if (updateArtifact != null) {
+                // Compare versions
+                String installedVersion = installedManifest.getVersion();
+                String availableVersion = updateArtifact.getVersion();
+
+                if (installedVersion.compareTo(availableVersion) < 0) {
+                    System.out.println("Upgrade available for " + installedManifest.getArtifactId() + ": " + installedVersion + " -> " + availableVersion);
+                } else if (installedVersion.compareTo(availableVersion) > 0) {
+                    System.out.println("Downgrade detected for " + installedManifest.getArtifactId() + ": " + installedVersion + " -> " + availableVersion);
```

Comment:
I think the comparison logic here should not be an alphabetic comparison, but something that understands semantic versioning.

The problem is that e.g. version "1.11" would be evaluated as lower than "1.2" when using alphabetical ordering, while it should  be higher semantically.

Reply:
Thanks for noticing it @TomasHofman 
I have the code where I am using DefaultArtifactVersion which we compare the versions semantically.

Reply:
`org.wildfly.channel.version.VersionMatcher` has the correct version compare logic